File: api-core/src/main/java/io/nerv/core/mvc/vo/TreeSelectVo.java
package io.nerv.core.mvc.vo;
import lombok.Data;

import java.util.List;

/**
 * antd treeSelect需要的数据结构
 * @author: S.PKAQ
 * @Datetime: 2019/3/19 21:45
 */
@Data
public class TreeSelectVo {
    /** 全树唯一 **/
    private String key;
    /** 显示名称 **/
    private String title;
    /** 节点值 **/
    private String value;
    /** 是否叶子 **/
    private boolean isLeaf;
    /** 路径 **/
    private String path;
    /** 路径中文描述 **/
    private String pathName;
    /** 上级节点id **/
    private String parentId;
    /** 上级节点名称 **/
    private String parentName;
    /** 子节点 **/
    private List<TreeSelectVo> children;
}

File: eva-core/src/main/kotlin/io/nerv/core/mvc/vo/TreeSelectVo.kt
package io.nerv.core.mvc.vo

/**
 * antd treeSelect需要的数据结构
 * @author: S.PKAQ
 * @Datetime: 2019/3/19 21:45
 */
open class TreeSelectVo {
    /** 全树唯一  */
    val key: String? = null

    /** 显示名称  */
    val title: String? = null

    /** 节点值  */
    val value: String? = null

    /** 是否叶子  */
    val isLeaf = false

    /** 路径  */
    val path: String? = null

    /** 路径中文描述  */
    val pathName: String? = null

    /** 上级节点id  */
    val parentId: String? = null

    /** 上级节点名称  */
    val parentName: String? = null

    /** 子节点  */
    val children: List<TreeSelectVo>? = null
}